libnetwork/iptables/firewalld.go

```diff
@@ -159,13 +182,23 @@ func OnReloaded(callback func()) {
 }
 
 // Call some remote method to see whether the service is actually running.
-func checkRunning() bool {
+func checkRunning(ctx context.Context) (bool, error) {
+	var err error
 	if connection == nil {
-		return false
+		return false, nil
+	}
+
+	var currentState string
+	err = connection.sysObj.CallWithContext(ctx, "org.freedesktop.DBus.Properties.Get", 0, dbusInterface, "state").Store(&currentState)
+
+	if err != nil {
+		return false, fmt.Errorf("failed to get current state of dbus connection: %w", err)
 	}
-	var zone string
-	err := connection.sysObj.Call(dbusInterface+".getDefaultZone", 0).Store(&zone)
-	return err == nil
+	// Running and failed are treated the same as we only want to verify if Firewalld is up and accepting dbus RPC calls
+	// If Firewalld is in INIT false is returned as there were observed failure cases where dbus calls did not succeed
+	// after a timeout.  Its safer to assume INIT should return false
+	log.G(ctx).Debugf("Firewalld's current reported state is, %s", currentState)
```

Comment:
Looks like there's a stray comma here;

```suggestion
	log.G(ctx).Debugf("Firewalld's current reported state is %s", currentState)
```

(or perhaps a `:`);

```suggestion
	log.G(ctx).Debugf("Firewalld's current reported state: %s", currentState)
```